common/include/pcl/pcl_macros.h

```diff
@@ -67,24 +67,53 @@
 
 #include <pcl/pcl_config.h>
 
+#include <boost/preprocessor/arithmetic/sub.hpp>
+#include <boost/preprocessor/comparison/equal.hpp>
+#include <boost/preprocessor/comparison/greater.hpp>
+#include <boost/preprocessor/control/if.hpp>
+
 // It seems that __has_cpp_attribute doesn't work correctly
 // when compiling with some versions of nvcc so we
 // additionally check if nvcc is used before setting the
-// PCL_DEPRECATED macro to [[deprecated]].
+// PCL_DEPRECATED_IMPL macro to [[deprecated]].
 #if defined(__has_cpp_attribute) && __has_cpp_attribute(deprecated) && !defined(__CUDACC__)
-  #define PCL_DEPRECATED(message) [[deprecated(message)]]
+  #define PCL_DEPRECATED_IMPL(message) [[deprecated(message)]]
 #elif defined(__GNUC__) || defined(__clang__)
-  #define PCL_DEPRECATED(message) __attribute__((deprecated(message)))
+  #define PCL_DEPRECATED_IMPL(message) __attribute__((deprecated(message)))
 #elif defined(_MSC_VER)
   // Until Visual Studio 2013 you had to use __declspec(deprecated).
```

Comment:
I believe we can remove this comment. In general why not make it `#define PCL_DEPRECATED(message) [[deprecated(message)]]` for all compilers? We require C++14 and this is [supported](https://en.cppreference.com/w/cpp/compiler_support) by any relevant C++14 compiler.

Reply:
`nvcc` is sadly still relevant

Reply:
Isn't it possible then to change the conditions so, that nvcc get's the old flags, but the other compilers are using the new attribute? Currently no compiler will use `[[deprecated(message)]]` as we force the compiler to allow only C++14 code, but `__has_cpp_attribute` is C++20.

Reply:
> but `__has_cpp_attribute` is C++20.

[No, it's C++14](https://en.cppreference.com/w/User:D41D8CD98F/feature_testing_macros#C.2B.2B14)

Reply:
Ah sorry, just saw the headline [here](https://en.cppreference.com/w/cpp/feature_test).

Reply:
Please unresolve this if your concerns weren't addressed. 😄